File: 品味Java的21種設計模式/dp_cpp/proxy/example4/Order.cpp
#include "Order.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace proxy
			{
				namespace example4
				{

					Order::Order(std::string productName, int orderNum, std::string orderUser)
					{
						this->productName = productName;
						this->orderNum = orderNum;
						this->orderUser = orderUser;
					}

					std::string Order::getProductName()
					{
						return productName;
					}

					void Order::setProductName(std::string productName, std::string user)
					{
						this->productName = productName;
					}

					int Order::getOrderNum()
					{
						return orderNum;
					}

					void Order::setOrderNum(int orderNum, std::string user)
					{
						this->orderNum = orderNum;
					}

					std::string Order::getOrderUser()
					{
						return orderUser;
					}

					void Order::setOrderUser(std::string orderUser, std::string user)
					{
						this->orderUser = orderUser;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/proxy/example6/Order.java
package cn.javass.dp.proxy.example6;
/**
 * 订单对象
 */
public class Order{
	/**
	 * 订单订购的产品名称
	 */
	private String productName;
	/**
	 * 订单订购的数量
	 */
	private int orderNum;
	/**
	 * 创建订单的人员
	 */
	private String orderUser;
	
	/**
	 * 构造方法，传入构建需要的数据
	 * @param productName 订单订购的产品名称
	 * @param orderNum 订单订购的数量
	 * @param orderUser 创建订单的人员
	 */
	public Order(String productName,int orderNum,String orderUser){
		this.productName = productName;
		this.orderNum = orderNum;
		this.orderUser = orderUser;
	}
	
	public String getProductName() {
		return productName;
	}
	public void setProductName(String productName,String user) {
		this.productName = productName;
	}
	public int getOrderNum() {
		return orderNum;
	}
	public void setOrderNum(int orderNum,String user) {
		this.orderNum = orderNum;
	}
	public String getOrderUser() {
		return orderUser;
	}
	public void setOrderUser(String orderUser,String user) {
		this.orderUser = orderUser;
	}
}
